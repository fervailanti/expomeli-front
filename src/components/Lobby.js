import React from 'react'

// This is just a kind of readme in home route...

const emoji = (emoji) => (
  <span role="img" aria-label="emoji">
    {emoji}
  </span>
)

const L = ({ to, children }) => (
  <a href={to} target="_blank" className="text-blue-400" rel="noopener noreferrer">
    {children}
  </a>
)

const Lobby = () => {
  return (
    <section className="layout">
      <div className="lobby">
        <div className="mb-4">
          <h1>Â¡Hola! {emoji('ðŸ‘‹')}</h1>
          <p>Ya puedes empezÃ¡r a encontrar productos en el buscardor de acÃ¡ arriba {emoji('ðŸ‘†')}</p>
          <p className="mb-4">Existen dos rutas mÃ¡s (aparte de la actual):</p>
          <ul className="mb-5">
            <li>
              <code>/items?search=</code> para motrar los resultados de la bÃºsqueda.
            </li>
            <li>
              <code>items/:id</code> para mostrar el detalle de un producto puntual seleccionado.
            </li>
          </ul>
          <p className="font-thin">
            <b>Nota</b>: Al ser un proyecto con finalidad de test, <i>no permite la compra</i>, y
            solo mostrarÃ¡n 4 productos por bÃºsqueda. AdemÃ¡s, no se montrarÃ¡n todos los detalles
            disponibles de los productos, solo los mÃ¡s destacados de estos.
          </p>
        </div>
        <div>
          <h2>Stack</h2>
          <p className="mb-4">
            <b>_Frontend</b>: estÃ¡ construido esencialmente con{' '}
            <L to="https://es.reactjs.org/">React</L> (Hooks),{' '}
            <L to="https://reactrouter.com/">React Router</L>,{' '}
            <L to="https://tailwindcss.com/">TailwindCSS</L>,{' '}
            <L to="https://github.com/axios/axios">Axios</L>,{' '}
            <L to="https://testing-library.com/docs/react-testing-library/intro">Testing Library</L>{' '}
            y un par mÃ¡s de paquetes detallados en el <code>package.json</code>.
          </p>
          <p>
            <b>_BFF (Backend For Frontend)</b>: estÃ¡ hecho con <L to="https://nodejs.org/">Node</L>{' '}
            y entorno a <L to="https://expressjs.com/">Express</L>. Se encarga principalmente de
            devolver la data preparada para su consumo en el Frontend. Tiene comunicaciÃ³n con las
            siguientes API's de MercadoLibre:
          </p>
          <ul className="mt-4 break-words">
            <li>
              <L to="https://api.mercadolibre.com/sites/MLA/search?q=â€‹:query">
                https://api.mercadolibre.com/sites/MLA/search?q=â€‹:query
              </L>
            </li>
            <li>
              <L to="https://api.mercadolibre.com/items/â€‹:id">
                https://api.mercadolibre.com/items/â€‹:id
              </L>
            </li>
            <li>
              <L to="https://api.mercadolibre.com/items/â€‹:idâ€‹/description">
                https://api.mercadolibre.com/items/â€‹:idâ€‹/description
              </L>
            </li>
          </ul>
        </div>
        <div className="mb-3">
          <h2>Autor</h2>
          <p>
            Desarrollado por <b>Fernando Vailanti</b> en Buenos Aires, Argentina.
          </p>
          <p>
            Puedes encontrarme en <L to="https://github.com/fervailanti">GitHub</L>,{' '}
            <L to="https://www.linkedin.com/in/fernando-vailanti-559b96164/">LinkedIn</L>,{' '}
            <L to="https://medium.com/@fernandovailanti">Medium</L> y{' '}
            <L to="mailto:fernandovailanti17@gmail.com">Gmail</L>.
          </p>
          <p>Â¡Saludos! {emoji('ðŸ’ª')}</p>
        </div>
      </div>
    </section>
  )
}

export default Lobby
