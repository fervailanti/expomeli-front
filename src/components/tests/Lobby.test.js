import React from 'react'
import Lobby from '../Lobby'
import { render } from '@testing-library/react'

it('Lobby component should render correctly', () => {
  const tree = render(<Lobby />)
  expect(tree.container).toMatchSnapshot()
})
