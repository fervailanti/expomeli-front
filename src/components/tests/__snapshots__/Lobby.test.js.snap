// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lobby component should render correctly 1`] = `
<div>
  <section
    class="layout"
  >
    <div
      class="lobby"
    >
      <div
        class="mb-4"
      >
        <h1>
          Â¡Hola! 
          <span
            aria-label="emoji"
            role="img"
          >
            ðŸ‘‹
          </span>
        </h1>
        <p>
          Ya puedes empezÃ¡r a encontrar productos en el buscardor de acÃ¡ arriba 
          <span
            aria-label="emoji"
            role="img"
          >
            ðŸ‘†
          </span>
        </p>
        <p
          class="mb-4"
        >
          Existen dos rutas mÃ¡s (aparte de la actual):
        </p>
        <ul
          class="mb-5"
        >
          <li>
            <code>
              /items?search=
            </code>
             para motrar los resultados de la bÃºsqueda.
          </li>
          <li>
            <code>
              /items/:id
            </code>
             para mostrar el detalle de un producto puntual seleccionado.
          </li>
        </ul>
        <p
          class="font-thin"
        >
          <b>
            Nota
          </b>
          : Al ser un proyecto con finalidad de test, 
          <i>
            no permite la compra
          </i>
          , y solo se mostrarÃ¡n 4 productos por bÃºsqueda. AdemÃ¡s, no se montrarÃ¡n todos los detalles disponibles de los productos, solo los mÃ¡s destacados de estos.
        </p>
      </div>
      <div>
        <h2>
          Stack
        </h2>
        <p
          class="mb-4"
        >
          <a
            class="text-blue-400"
            href="https://github.com/fervailanti/expomeli-front"
            rel="noopener noreferrer"
            target="_blank"
          >
            <b>
              _Frontend
            </b>
          </a>
          : estÃ¡ construido esencialmente con 
          <a
            class="text-blue-400"
            href="https://es.reactjs.org/"
            rel="noopener noreferrer"
            target="_blank"
          >
            React
          </a>
           (Hooks),
           
          <a
            class="text-blue-400"
            href="https://reactrouter.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            React Router
          </a>
          ,
           
          <a
            class="text-blue-400"
            href="https://tailwindcss.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            TailwindCSS
          </a>
          ,
           
          <a
            class="text-blue-400"
            href="https://github.com/axios/axios"
            rel="noopener noreferrer"
            target="_blank"
          >
            Axios
          </a>
          ,
           
          <a
            class="text-blue-400"
            href="https://testing-library.com/docs/react-testing-library/intro"
            rel="noopener noreferrer"
            target="_blank"
          >
            Testing Library
          </a>
           
          y un par mÃ¡s de paquetes detallados en el 
          <code>
            package.json
          </code>
          .
        </p>
        <p>
          <a
            class="text-blue-400"
            href="https://github.com/fervailanti/expomeli-api"
            rel="noopener noreferrer"
            target="_blank"
          >
            <b>
              _API (Backend For Frontend)
            </b>
          </a>
          : estÃ¡ hecho principalmente en 
          <a
            class="text-blue-400"
            href="https://nodejs.org/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Node.js
          </a>
           y
           
          <a
            class="text-blue-400"
            href="https://expressjs.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Express
          </a>
          , ademÃ¡s de
           
          <a
            class="text-blue-400"
            href="https://github.com/standard-things/esm#readme"
            rel="noopener noreferrer"
            target="_blank"
          >
            ES Modules
          </a>
           y
           
          <a
            class="text-blue-400"
            href="https://github.com/axios/axios"
            rel="noopener noreferrer"
            target="_blank"
          >
            Axios
          </a>
          . Se encarga principalmente de manejar la informaciÃ³n y prepararla para su consumo en el Frontend. Tiene comunicaciÃ³n con los siguientes endpoints de MercadoLibre:
        </p>
        <ul
          class="mt-4 break-words"
        >
          <li>
            <a
              class="text-blue-400"
              href="https://api.mercadolibre.com/sites/MLA/search?q=â€‹:query"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://api.mercadolibre.com/sites/MLA/search?q=â€‹:query
            </a>
          </li>
          <li>
            <a
              class="text-blue-400"
              href="https://api.mercadolibre.com/items/â€‹:id"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://api.mercadolibre.com/items/â€‹:id
            </a>
          </li>
          <li>
            <a
              class="text-blue-400"
              href="https://api.mercadolibre.com/items/â€‹:idâ€‹/description"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://api.mercadolibre.com/items/â€‹:idâ€‹/description
            </a>
          </li>
        </ul>
      </div>
      <div
        class="mb-3"
      >
        <h2>
          Autor
        </h2>
        <p>
          Desarrollado por 
          <b>
            Fernando Vailanti
          </b>
           en Buenos Aires, Argentina 
          <span
            aria-label="emoji"
            role="img"
          >
            ðŸ‡¦ðŸ‡·
          </span>
        </p>
        <p>
          Puedes encontrarme en 
          <a
            class="text-blue-400"
            href="https://github.com/fervailanti"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          ,
           
          <a
            class="text-blue-400"
            href="https://www.linkedin.com/in/fernando-vailanti-559b96164/"
            rel="noopener noreferrer"
            target="_blank"
          >
            LinkedIn
          </a>
          ,
           
          <a
            class="text-blue-400"
            href="https://medium.com/@fernandovailanti"
            rel="noopener noreferrer"
            target="_blank"
          >
            Medium
          </a>
           y
           
          <a
            class="text-blue-400"
            href="mailto:fernandovailanti17@gmail.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Gmail
          </a>
          .
        </p>
        <p>
          Â¡Saludos! 
          <span
            aria-label="emoji"
            role="img"
          >
            ðŸ’ª
          </span>
        </p>
      </div>
    </div>
  </section>
</div>
`;
